src/jdk.jpackage/share/native/common/tstrings.h

```diff
@@ -356,6 +356,11 @@ namespace tstrings {
             data << fromUtf8(msg);
         }
 
+        any& operator << (const std::string& msg) {
+            data << fromUtf8(msg);
+            return *this;
+        }
+
```

Comment:
I guess you also need 

        any& operator << (LPSTR v) {
            data << (v ? fromUtf8(v) : "NULL");
            return *this;
        }

        any& operator << (LPCSTR v) {
            data << (v ? fromUtf8(v) : "NULL");
            return *this;
        }

to make a complete set of overloads

Reply:
I deliberately left those 2 out since the comments accompanying them seem to suggest that they're Windows specific overloads that are there for different reasons (As opposed to the one this PR is aimed at, which takes in a regular std::wstring), moreover not providing overloads for them don't seem to be causing issues either. I'm unsure if adding them too would be correct, I'll wait for more reviews before doing so

Reply:
I'm the author of these comments and basically the entire source file.

You are right, it will work without the additional overloads. However the null pointers will be ignored (tested with G++ 7.4.0), but this is minor, probably implementation-specific and can be ignored.

Reply:
Strange, on my end gcc doesn't mind NULL being passed (though this may be due to a different gcc version being used), guess it's probably implementation specific like you mentioned

Reply:
Correct, it allows NULL to be passed. G++ 7.4.0 ignores it and doesn't append anything to the buffer, while msvc back when this code was created appended a "null" string to the buffer.